test/framework/src/main/java/org/elasticsearch/cluster/coordination/LinearizabilityChecker.java

```diff
@@ -207,15 +213,10 @@ public String toString() {
     }
 
     /**
-     * Checks whether the provided history is linearizable with respect to the given sequential specification
-     *
-     * @param spec the sequential specification of the datatype
-     * @param history the history of events to check for linearizability
-     * @param missingResponseGenerator used to complete the history with missing responses
-     * @return true iff the history is linearizable w.r.t. the given spec
+     * Convenience method for {@link #isLinearizable(SequentialSpec, History, Function)} that requires the history to be complete
      */
-    public static boolean isLinearizable(SequentialSpec spec, History history, Function<Object, Object> missingResponseGenerator) {
-        return isLinearizable(spec, history, missingResponseGenerator, () -> false);
+    public static boolean isLinearizable(SequentialSpec spec, History history) throws LinearizableCheckAborted {
+        return isLinearizable(spec, history, o -> { throw new IllegalArgumentException("history is not complete"); });
```

Comment:
Should this be an assertion error rather than an IAE?

Reply:
Moved as is, but I agree, this should be changed to AssertionError